methods.js

```diff
@@ -1,138 +1,66 @@
-    })
-}
-
-// Llamada de la función principal (handler)
-const getMoviesByTitle = (title) =>{
-
-    // funcion principal (handler) encargada gestionar las peticiones y devolvewr las respuestas finales
-    // Por tanto, tendréis que crearla y llamarla desde la ruta correspondiente del servidor y devolver la respuesta en el callback
-    
-    getMoviesFromMoviesDataByTitle(title)
-    .then(moviesMatched=>{
-        console.log(moviesMatched)
-        return moviesMatched
-    } )
-    .catch(error=>{
-        console.log(error)
-        return error
-    })
-}
-
-// Llamada de la función principal (handler):
-getMoviesByTitle("The S")
-// Llamada que devuelve error:
-// getMoviesByTitle("X")
-
-
-
-
-//-----------------------------//
-//   3.  getMovieByShowtimes   //
-//-----------------------------//
-
-const getMoviesFromMoviesDataByShowtime = (showTime) => {
-// función encargada de la petición a la "base de datos".
-// esta función será exportada desde este archivo. 
-
-    return new Promise( ( resolve, reject ) => {
-        setTimeout(()=> {
-
-            const moviesMatched = movies.filter( ( movie )=> movie.showtimes.includes(showTime) )
-
-            if (!moviesMatched || moviesMatched.length ===0) reject("No hay coincidencias")
-            
-            resolve(moviesMatched)
-
-        },2000)
-    })
-
-}
-
-const getMoviesByShowtime = async (swhotime) => {
-// funcion principal (handler) encargada gestionar las peticiones y devolvewr las respuestas finales
-// Por tanto, tendréis que crearla y llamarla desde la ruta correspondiente del servidor y devolver la respuesta en el callback
-
-    try{
-
-        const moviesMatched = await getMoviesFromMoviesDataByShowtime(swhotime)
-        
-        console.log(moviesMatched)
-        return moviesMatched
-
-    } catch(error) {
-
-        console.log(error)
-        return  error
-
-    }
-}
-
-
-// Llamada de la función principal (handler)
-getMoviesByShowtime("13:50")
-// Llamada que devuelve error:
-// getMoviesByShowtime("20:33")
+const movies = require("./moviesData");
+
+// getMovies
+
+const getMovies = callback => {
+    console.log("Getting movies");
+    setTimeout(() => {
+        if (!movies.movies) callback(`Not exist a movies`);
+        else callback(null, movies.movies);
+    }, 3000);
+};
+
+// getMovieById
+
+const getMovieById = (id, callback) => {
+    console.log("Initializating Movie search by ID", id);
+    setTimeout(() => {
+        const movie = movies.movies.find(movie => {
+            return movie.id === id;
+        });
+        if (!movie) callback(`Opss - ID movie ${id} not found`);
+        else callback(null, movie);
+    }, 3000);
+};
+
+// getMovieByTitle
+
+const getMovieByTitle = async title => {
+    console.log("Initializating Movie search by Title", title);
+    return new Promise((resolve, reject) => {
+        setTimeout(() => {
+            let matches = [];
+            for (let i = 0; i < movies.movies.length; i++) {
```

Comment:
Me gusta que lo hayas hecho con un bucle for, es clásico pero efectivo. 

Solo como sugerencia: es cierto que actualmente se suele evitar usar los bucles _for_ por razones principalmente semánticas: cuando la operación se complica un poco, es mucho más fácil de interpretar para otro programador si utilizas el método filter, que es usado precisamente para este caso de uso. Reduces código y de un solo vistazo se puede ver qué está ocurriendo. 

Aún así, como te digo, no es un error usar bucles for. Simplemente, en la actualidad, te encontrarás con más funciones como filter, map... que directamente el bucle for. 

Otra posibilidad, si te gustan los for, es usar el método _for ... of_. Es una revisión del bucle _for_ para estos casos de uso. Si quieres, puedes echarle un vistazo por MDN: https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Sentencias/for...of